src/Illuminate/Database/Eloquent/Concerns/HasRelationships.php

```diff
@@ -565,14 +565,14 @@ protected function guessBelongsToManyRelation()
      * @param  string  $related
      * @return string
      */
-    public function joiningTable($related)
+    public function joiningTable($related, $instance = null)
```

Comment:
Is this a breaking change if someone is already overriding this method?

Reply:
Yes. It's a change of a method signature in the public API.

Since it's a trait, you won't get incompatible declaration exceptions, but you also won't have any warning if, for example, you're overriding this method to add more parameters, and calling `belongsToMany()` now passes a parameter you weren't expecting into it.

That said, I'm finding it hard to come up with a reasonable example to demonstrate this. @sisve usually has a better eye for these things.

Reply:
You're right. Any change of a public api signature is a breaking change. In this case the user will see weird behavior if mixing packages that targets different 5.6 versions. If the package that override joiningTable is modeled after 5.6.0, then the user will never see their joiningTableSegment method called and there will be questions about it.